tests/integration/__init__.py

```diff
@@ -170,6 +236,294 @@ def run_testcase(self, testcase, needs_daemon=True):  # pylint: disable=W0221
+                            script_name
+                        )
+                    )
+                sfh.write(
+                    '#!{0}\n'.format(sys.executable) +
+                    '\n'.join(script_template).format(script_name.replace('salt-', ''))
+                )
+            fst = os.stat(script_path)
+            os.chmod(script_path, fst.st_mode | stat.S_IEXEC)
+
+        return script_path
+
+    def get_script_args(self):  # pylint: disable=no-self-use
+        '''
+        Returns any additional arguments to pass to the CLI script
+        '''
+        return []
+
+
+class SaltDaemonScriptBase(SaltScriptBase):
+    '''
+    Base class for Salt Daemon CLI scripts
+    '''
+
+    def __init__(self, *args, **kwargs):
+        super(SaltDaemonScriptBase, self).__init__(*args, **kwargs)
+        self._running = multiprocessing.Event()
+        self._connectable = multiprocessing.Event()
+        self._process = None
+
+    def is_alive(self):
+        '''
+        Returns true if the process is alive
+        '''
+        return self._running.is_set()
+
+    def get_check_ports(self):  # pylint: disable=no-self-use
+        '''
+        Return a list of ports to check against to ensure the daemon is running
+        '''
+        return []
+
+    def start(self):
+        '''
+        Start the daemon subprocess
+        '''
+        self._process = SignalHandlingMultiprocessingProcess(
+            target=self._start, args=(self._running,))
+        self._process.start()
+        self._running.set()
+        return True
+
+    def _start(self, running_event):
+        '''
+        The actual, coroutine aware, start method
+        '''
+        log.info('Starting pytest %s DAEMON', self.__class__.__name__)
+        proc_args = [
+            self.get_script_path(self.cli_script_name),
+            '-c',
+            self.config_dir,
+        ] + self.get_script_args()
+        log.info('Running \'%s\' from %s...', ' '.join(proc_args), self.__class__.__name__)
+
+        try:
+            terminal = NonBlockingPopen(proc_args,
+                                        cwd=CODE_DIR,
+                                        env={'PYTHONPATH': CODE_DIR})
+            self.pid = terminal.pid
+
+            while running_event.is_set() and terminal.poll() is None:
+                # We're not actually interested in processing the output, just consume it
+                if terminal.stdout is not None:
+                    terminal.recv()
+                if terminal.stderr is not None:
+                    terminal.recv_err()
+                time.sleep(0.125)
+        except (SystemExit, KeyboardInterrupt):
+            # Let's close the terminal now that we're done with it
+            try:
+                terminal.send_signal(signal.SIGTERM)
+            except OSError as exc:
+                if exc.errno != errno.ESRCH:
+                    raise
+            terminal.communicate()
```

Comment:
This should be changed to:

```
if exc.errno not in (signal.ESRCH, signal.EACCES):
    raise
```

The reason is that on Windows, if the process doesn't exist, it will return `OSError` with errno of `EACCES` instead of `ESRCH`. I have already made similar changes in `salt/utils/process.py`.